output/cloud/expv2/collect.go

```diff
@@ -0,0 +1,153 @@
+package expv2
+
+import (
+	"errors"
+	"sync"
+	"time"
+
+	"go.k6.io/k6/metrics"
+	"go.k6.io/k6/output"
+)
+
+type bucketQ struct {
+	m       sync.Mutex
+	buckets []timeBucket
+}
+
+// Buckets pop all the queued buckets.
+func (q *bucketQ) Buckets() []timeBucket {
+	q.m.Lock()
+	b := q.buckets
+	q.buckets = q.buckets[:0]
```

Comment:
I haven't grokked the reminder of this code yet, but this part seems problematic, since `b` and `q.buckets` will reference the same underlying array, and I can't think of a situation where this would be intended and we'd still need locking here.

Reply:
https://github.com/grafana/k6/pull/3071/commits/5db6157071e6bcea7e7c3f822a4cbe9dbab50294 applied an improvement for it